### ISSUE TITLE
toString on Typeable is not helpful

### ISSUE THREAD

**Issue by rorygraves** [milessabin/shapeless#313]
In Ensime we are using shapeless to do typesafe calls to an actor:

```scala
  def callRPC[R](target: ActorRef, request: RPCRequest, maxWait: FiniteDuration = defaultMaxWait)(implicit typ: Typeable[R]): R = {
    val future = target.ask(request)(maxWait)
    val result = Await.result(future, maxWait)
    result match {
      case e: RPCError =>
        throw e
      case r =>
        val castRes = r.cast[R]
        if (castRes.isDefined)
          castRes.get
        else {
          val msg = "Error, incorrect type found in RPC call, expected: " + typ + " got " + r.getClass
          val ex = RPCError(ProtocolConst.ErrExceptionInRPC, msg)
          log.error(msg, ex)
          throw ex
        }
    }
  }
```

I'm sure there are many things that you dislike about this code, but my problem is I cannot generate a good error message: - on failure I get:

```
org.ensime.core.RPCError: 201: Error, incorrect type found in RPC call, expected: shapeless.LowPriorityTypeable$$anon$1@7383291a got class java.lang.Boolean```
```
Is there any way to improve the ```toString``` to include the expected type (```R``` in this case)?

e..g. Something like```Typeable(Boolean)``` or/and add a seperate method to get the type directly.

**Comment by milessabin** [comment]
Reasonable request. It won't get in to 2.1.0 I'm afraid though (at least not unless a perfectly formed PR lands in my inbox very very soon).

**Comment by stacycurl** [comment]
For some definition of 'soon' ;p

**Comment by milessabin** [comment]
Oh _very_ funny ;-)

**Comment by rorygraves** [comment]
:smiley:

**Comment by milessabin** [comment]
Fixed in #335.
